module/Photo/src/Photo/Service/AlbumCover.php

```diff
@@ -0,0 +1,158 @@
+<?php
+
+namespace Photo\Service;
+
+use Application\Service\AbstractService;
+use Zend\ServiceManager\ServiceManager;
+use Zend\ServiceManager\ServiceManagerAwareInterface;
+use Photo\Model\Album as AlbumModel;
+use Photo\Model\Photo as PhotoModel;
+use Imagick;
+
+/**
+ * Album cover services. Used for (re)generating album covers.
+ * 
+ */
+class AlbumCover extends AbstractService
+{
+
+    public function createCover($album)
+    {
+        $cover = $this->generateCover($album);
+        $tempFileName = sys_get_temp_dir() . '/CoverImage' . rand() . '.png';
+        $cover->writeImage($tempFileName);
+        $newPath = $this->getPhotoService()->generateStoragePath($tempFileName);
+        $config = $this->getConfig();
+        rename($tempFileName, $config['upload_dir'] . '/' . $newPath);
+        return $newPath;
+    }
+
+    protected function generateCover($album)
+    {
+        $config = $this->getConfig();
+        $columns = $config['album_cover']['cols'];
+        $rows = $config['album_cover']['rows'];
+        $count = $columns * $rows;
+        $images = $this->getImages($album, $count);
+        //if there are not enough images to fill the matrix, reduce the rows and columns
+        while (($columns > 1 || $rows > 1) && count($images) < $count) {
+            if ($columns < $rows) {
+                $rows--;
+            } else {
+                $columns--;
+            }
+            $count = $rows * $columns;
+        }
+        $innerWidth = $config['album_cover']['width'] - 2 * $config['album_cover']['outer_border'];
+        $innerHeight = $config['album_cover']['height'] - 2 * $config['album_cover']['outer_border'];
+        $imageWidth = floor(($innerWidth - ($columns - 1) * $config['album_cover']['inner_border']) / $columns);
+        $imageHeight = floor(($innerHeight - ($rows - 1) * $config['album_cover']['inner_border']) / $rows);
+        //increase outer border due to flooring of image dimensions
+        $outerBorderX = $config['album_cover']['outer_border'] + ceil(($innerWidth - ($columns * $imageWidth + ($columns - 1) * $config['album_cover']['inner_border'])) / 2);
+        $outerBorderY = $config['album_cover']['outer_border'] + ceil(($innerHeight - ($rows * $imageHeight + ($rows - 1) * $config['album_cover']['inner_border'])) / 2);
+        // Make a blank canvas
+        $target = new Imagick();
```

Comment:
Voor native PHP classes een backslash er voor, en niet de namespace includen.

I.e.
use Imagick();
(..)
new Imagick()

word dan gewoon New \Imagick(); Dit geeft duidelijker aan dat het een native php versie is, en scheelt weer code en ongebruikte imports later in een project